pkg/api/defaults.go

```diff
@@ -424,7 +418,7 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 
 		// Upgrade scenario:
 		// We need to force set EnableRbac to true for upgrades to 1.15.0 and greater if it was previously set to false (AKS Engine only)
-		if !a.OrchestratorProfile.KubernetesConfig.IsRBACEnabled() && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.15.0") && isUpgrade && !cs.Properties.IsHostedMasterProfile() {
+		if !a.OrchestratorProfile.KubernetesConfig.IsRBACEnabled() && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.15.0") && isUpgrade {
```

Comment:
Optionally, we could also drop the `common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.15.0")` test here. I guess I missed that in #3851.

Reply:
Yeah we should do a follow-up workstream *just* to (1) get rid of the unnecessary version stuff, and (2) automate these versions in tests so we don't have to ever again manually maintain them